pages/userSettings/userSettings.js

```diff
@@ -0,0 +1,72 @@
+const ERR = require('async-stacktrace');
+const express = require('express');
+const router = express.Router();
+const crypto = require('crypto');
+const uuidv4 = require('uuid/v4');
+
+const error = require('@prairielearn/prairielib/error');
+const sqldb = require('@prairielearn/prairielib/sql-db');
+const sqlLoader = require('@prairielearn/prairielib/sql-loader');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+router.post('/', (req, res, next) => {
+    if (req.body.__action === 'token_generate') {
+        const name = req.body.token_name;
+        const token = uuidv4();
+        const tokenHash = crypto.createHash('sha256').update(token, 'utf8').digest('hex');
+
+        const params = {
+            user_id: res.locals.authn_user.user_id,
+            name,
+            // The token will only be persisted until the next page render
+            // After that, we'll remove it from the database
+            token,
+            token_hash: tokenHash,
+        };
+        sqldb.queryOneRow(sql.insert_access_token, params, (err) => {
```

Comment:
Can we insert a row into `audit_logs` whenever we do anything with tokens? For example, insert like `https://github.com/PrairieLearn/PrairieLearn/blob/master/sprocs/assessment_instances_update.sql#L73` or delete like `https://github.com/PrairieLearn/PrairieLearn/blob/master/sprocs/assessment_instances_delete.sql#L31`. We don't use `audit_logs` for anything in the UI, but I've found it helpful in the past when tracking down bugs and also when figuring out what some weird user had been up to :-)

A pattern we were using elsewhere is to make a file like `sprocs/access_tokens_insert.sql` that does both the insert itself and the audit log update. I don't mind whether we continue that pattern or just call again from JS to update the audit log.